Mathlib/Algebra/Category/ModuleCat/Presheaf/ChangeOfRings.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Category.ModuleCat.ChangeOfRings
+import Mathlib.Algebra.Category.ModuleCat.Presheaf
+
+/-!
+# Change of presheaf of rings
+
+In this file, we define the restriction of scalars functor
+`restrictScalars α : PresheafOfModules.{v} R' ⥤ PresheafOfModules.{v} R`
+attached to a morphism of presheaves of rings `α : R ⟶ R'`.
+
+-/
+
+universe v v' u u'
+
+open CategoryTheory
+
+namespace PresheafOfModules
+
+variable {C : Type u'} [Category.{v'} C] {R R' : Cᵒᵖ ⥤ RingCat.{u}}
+
+/-- The restriction of scalars of presheaves of modules, on objects. -/
+@[simps]
+noncomputable def restrictScalarsBundledCore (M' : PresheafOfModules R') (α : R ⟶ R') :
+    BundledCorePresheafOfModules R where
+  obj X := (ModuleCat.restrictScalars (α.app X)).obj (M'.obj X)
+  map {X Y} f :=
+    { toFun := M'.map f
+      map_add' := map_add _
+      map_smul' := fun r x ↦ by
+        have eq := RingHom.congr_fun (α.naturality f) r
+        apply (M'.map_smul f (α.app _ r) x).trans
```

Comment:
```suggestion
        erw [(M'.map_smul f (α.app _ r) x)]
```
I haven't tested this locally yet, but maybe it's a good idea to have this as a `erw` based on this [thread](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60erw.60.20vs.20.60apply.60).

Reply:
Okay based on the continuation of the discussion, feel free to ignore this